gptme/chat.py

```diff
@@ -235,9 +235,6 @@ def step(
         if msg_response:
             yield msg_response.replace(quiet=True)
             yield from execute_msg(msg_response, confirm)
-    except KeyboardInterrupt:
-        clear_interruptible()
-        yield Message("system", "Interrupted")
     finally:
         clear_interruptible()
```

Comment:
I'm not sure if this is correct, I think the `clear_interruptible` is no longer called?

I think the `finally` clause might not run for the generator until the `next()` call.

Reply:
Nevermind, it's there in the except clause for that very reason.

Reply:
Hmm no it's probably not the same. Let me find something.

Reply:
I've updated the MR. To stay on the safe side, I've called `clear_interruptible` in the `execute_msg` function. Notice the `break` in the except clause. Before it was still calling subsequent tools if any after a user interruption :facepalm:

Reply:
We could also call `clear_interruptible` inside the `handle_keyboard_interrupt` itself just before raising the `KeyboardInterrupt` exception, what do you think?